src/main/frontend/handler.cljs

```diff
@@ -73,51 +73,48 @@
     (state/pub-event! [:instrument {:type :blocks/count
                                     :payload {:total (db/blocks-count)}}])))
 
-(defn store-schema!
-  []
-  (storage/set :db-schema (assoc db-schema/schema
-                                 :db/version db-schema/version)))
-
 (defn restore-and-setup!
-  [repos old-db-schema]
-  (-> (db/restore!
-       {:repos repos}
-       old-db-schema
-       (fn [repo]
-         (file-handler/restore-config! repo)))
-      (p/then
-       (fn []
-         ;; try to load custom css only for current repo
-         (ui-handler/add-style-if-exists!)
-
-         ;; install after config is restored
-         (shortcut/unlisten-all)
-         (shortcut/refresh!)
-
-         (cond
-           (and (not (seq (db/get-files config/local-repo)))
-                ;; Not native local directory
-                (not (some config/local-db? (map :url repos)))
-                (not (mobile-util/native-platform?)))
-           ;; will execute `(state/set-db-restoring! false)` inside
-           (repo-handler/setup-local-repo-if-not-exists!)
-
-           :else
-           (state/set-db-restoring! false))))
-      (p/then
-       (fn []
-         (js/console.log "db restored, setting up repo hooks")
-         (store-schema!)
-
-         (state/pub-event! [:modal/nfs-ask-permission])
-
-         (page-handler/init-commands!)
-
-         (watch-for-date!)
-         (file-handler/watch-for-current-graph-dir!)
-         (state/pub-event! [:graph/restored (state/get-current-repo)])))
-      (p/catch (fn [error]
-                 (log/error :exception error)))))
+  [repos]
+  (when-let [repo (or (state/get-current-repo) (:url (first repos)))]
+    (-> (db/restore! repo)
+        (p/then
+         (fn []
+           ;; try to load custom css only for current repo
+           (ui-handler/add-style-if-exists!)
+
+           (->
+            (p/do! (repo-config-handler/start {:repo repo})
+                   (when (config/global-config-enabled?)
+                        (global-config-handler/start {:repo repo})))
+            (p/finally
+              (fn []
+                ;; install after config is restored
+                (shortcut/unlisten-all)
+                (shortcut/refresh!)
+
+                (cond
+                  (and (not (seq (db/get-files config/local-repo)))
+                       ;; Not native local directory
+                       (not (some config/local-db? (map :url repos)))
+                       (not (mobile-util/native-platform?)))
+                  ;; will execute `(state/set-db-restoring! false)` inside
+                  (repo-handler/setup-local-repo-if-not-exists!)
+
+                  :else
+                  (state/set-db-restoring! false)))))))
```

Comment:
I didn't look into the detailed cause. It seems iOS App will never clear db-restoring state now and get stuck at the splash screen.

Should test the following cases for mobile loading:

- Clear all App data, start from a fresh install
- Upgrade, all App data exists
- Changing App, all App data exits, but mobile documents path changed. This occurs when developing or switching install source from test flight to app-store.